quiche/src/h3/frame.rs

```diff
@@ -280,22 +289,133 @@ impl Frame {
+
+                if let Some(v) = qpack_blocked_streams {
+                    settings.push(qlog::events::h3::Setting {
+                        name: "QPACK_BLOCKED_STREAMS".to_string(),
+                        value: *v,
+                    });
+                }
+
+                if let Some(v) = h3_datagram {
+                    settings.push(qlog::events::h3::Setting {
+                        name: "H3_DATAGRAM".to_string(),
+                        value: *v,
+                    });
+                }
+
+                if let Some((k, v)) = grease {
+                    settings.push(qlog::events::h3::Setting {
+                        name: k.to_string(),
+                        value: *v,
+                    });
+                }
+
+                qlog::events::h3::Http3Frame::Settings { settings }
+            },
+
+            // Qlog expects the `headers` to be represented as an array of
+            // name:value pairs. At this stage, we only have the qpack block, so
+            // populate the field with an empty vec.
+            Frame::PushPromise { push_id, .. } => Http3Frame::PushPromise {
+                push_id: *push_id,
+                headers: vec![],
+            },
+
+            Frame::GoAway { id } => Http3Frame::Goaway { id: *id },
+
+            Frame::MaxPushId { push_id } =>
+                Http3Frame::MaxPushId { push_id: *push_id },
+
+            Frame::PriorityUpdateRequest {
+                prioritized_element_id,
+                priority_field_value,
+            } => Http3Frame::PriorityUpdate {
+                target_stream_type:
+                    qlog::events::h3::H3PriorityTargetStreamType::Request,
+                prioritized_element_id: *prioritized_element_id,
+                priority_field_value: String::from_utf8_lossy(
+                    priority_field_value,
+                )
+                .into_owned(),
+            },
+
+            Frame::PriorityUpdatePush {
+                prioritized_element_id,
+                priority_field_value,
+            } => Http3Frame::PriorityUpdate {
+                target_stream_type:
+                    qlog::events::h3::H3PriorityTargetStreamType::Request,
+                prioritized_element_id: *prioritized_element_id,
+                priority_field_value: String::from_utf8_lossy(
+                    priority_field_value,
+                )
+                .into_owned(),
+            },
+
+            Frame::Unknown {
+                raw_type,
+                payload_length,
+            } => Http3Frame::Unknown {
+                raw_frame_type: *raw_type,
+                raw_length: Some(*payload_length as u32),
+                raw: None,
+            },
+        }
+    }
 }
 
 impl std::fmt::Debug for Frame {
     fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
         match self {
-            Frame::Data { payload } => {
-                write!(f, "DATA len={}", payload.len())?;
```

Comment:
Why?

Reply:
Where the frames are logged, there is a change to include the payload length of every frame regardless of type. So this information is redundant.